src/compiler/crystal/syntax/parser.cr

```diff
@@ -1675,7 +1675,8 @@ module Crystal
       next_token_skip_space_or_newline
 
       name_location = @token.location
-      name = parse_ident allow_type_vars: false
+      name = parse_ident(allow_type_vars: false, parse_nilable: false)
+      raise "BUG: ModuleDef name can only be a Path" unless name.is_a?(Path)
```

Comment:
What do you think if we extract these two lines into a new `parse_path` method? The `raise` message can simply be "BUG: expected Path" because it never triggers in practice.

Reply:
(then of course we use `parse_path` everywhere else)

Reply:
Done, thanks